packages/sync-server/src/types.ts

```diff
@@ -23,10 +24,72 @@ export type SyncServerConfig = {
     perModelUpdateTimeoutMs: number;
     avoidRepull: boolean;
   };
+  snapshotTransactionTimeoutMs: number;
+  syncSessionTimeoutMs: number;
+  maxConcurrentSessions: number;
 };
 
 export interface SyncLookupQueryDetails {
   select?: string[];
   joins?: string[];
   where?: any;
 }
+
+export interface SnapshotParams {
+  since: number;
+  projectIds: string[];
+  deviceId: string;
+}
+
+export interface SessionIsProcessingResponse {
+  session_is_processing: boolean;
+}
+
+export interface GlobalClockResult {
+  tick: number;
+  tock: number;
+}
+
+export interface StartSessionResult {
+  sessionId: string;
+}
+
+export interface SyncSessionMetadata {
+  startedAtTick: number;
+}
+
+export interface PullMetadata {
+  totalToPull: number;
+  pullUntil: number;
+}
+
+export interface PrepareSessionResult {
+  sessionId: string;
+  tick: number;
+}
+
+export interface PullInitiationResult {
+  sessionId: string;
+  totalToPull: number;
+  pullUntil: number;
+}
+
+export type UnmarkSessionAsProcessingFunction = () => Promise<void>;
+
+export interface SyncSnapshotData {
+  id: number;
+  [key: string]: any;
+}
```

Comment:
Not sure how feasible this is, but since we have the luxury of working with new code:

```suggestion
export interface SyncSnapshotData<T> {
  id: number;
  [key: string]: T;
}
```

Or at least

```ts
interface SyncSnapshotData<T = unknown> // `any` if we really must
```

Reply:
Ah, sorry, I did misread the type! Perhaps, then, just:

```suggestion
export interface SyncSnapshotData {
  id: number;
  [key: string]: unknown;
}
```